src/casp/services/api/services/cell_operations_service.py

```diff
@@ -170,67 +170,71 @@ def __split_embeddings_into_chunks(embeddings: np.array, chunk_size: int) -> t.L
         num_chunks: int = math.ceil(len(embeddings) / chunk_size)
         return np.array_split(embeddings, num_chunks)
 
-    def get_cell_type_distribution(
+    async def annotate_adata_file(
         self,
-        query_ids: t.List[str],
-        knn_response: MatchResult,
+        user: models.User,
+        file: t.BinaryIO,
         model_name: str,
-        include_dev_metadata: bool,
-    ) -> t.List[t.Dict[str, t.Any]]:
-        """
-        Get cell type distribution for the given query ids and knn response. Put results into temporary table and
-        aggregate summary statistics.
-
-        :param query_ids: List of query ids (original cell ids from the input file).
-        :param knn_response: List of lists of MatchNeighbor objects returned by space vector search service.
-        :param model_name: Model name to use for matching.
-        :param include_dev_metadata: Boolean flag to include dev metadata in the response.
-
-        :return: List of cell type distribution objects.
-        """
-        cas_model = self.cellarium_general_dm.get_model_by_name(model_name=model_name)
-
-        temp_table_fqn = self.cell_operations_dm.insert_matches_to_temp_table(
-            query_ids=query_ids, knn_response=knn_response
-        )
-
-        if include_dev_metadata:
-            return self.cell_operations_dm.get_neighborhood_distance_summary_dev_details(
-                cas_model=cas_model, match_temp_table_fqn=temp_table_fqn
-            )
-
-        return self.cell_operations_dm.get_neighborhood_distance_summary(
-            cas_model=cas_model, match_temp_table_fqn=temp_table_fqn
-        )
-
-    async def annotate_adata_file(
-        self, user: models.User, file: t.BinaryIO, model_name: str, include_dev_metadata: bool
-    ) -> t.List[t.Dict[str, t.Any]]:
+        consensus_strategy: consensus_engine.ConsensusStrategyType,
+        include_dev_metadata: t.Optional[bool] = None,
+        normalize: t.Optional[bool] = None,
```

Comment:
(drop)

Reply:
instead, add **kwargs to allow passing strategy-specific parameters (like pruning thershold, prefactor).

Reply:
I'd like to avoid using `**kwargs` as it makes method arguments less specific. I will add threshold. What's a prefactor? What are other strategy parameters we can have?